src/clean.js

```diff
@@ -0,0 +1,93 @@
+const { contentfulManagementClient, contentfulPreviewClient, defaultLocale } = require('./config');
+
+const deleteEntry = async(id) => {
+  const contentfulManagementConnection = await contentfulManagementClient.connect();
+  let entry = await contentfulManagementConnection.getEntry(id).catch((e) => {
+    console.log(`couldn't load entry: ${id}\n` + e.toString());
+    return;
+  });
+
+  if (entry.sys.publishedVersion) {
+    console.log(`unpublish ${entry.sys.contentType.sys.id}`);
+    entry = await entry.unpublish().catch((e) => {
+      console.log('error unpublishing ' + JSON.stringify(e));
+    });
+  }
+
+  console.log(`delete ${entry.sys.contentType.sys.id}`);
+  entry.delete().catch(async(e) => {
+    console.log('error deleting'  + JSON.stringify(e));
+    throw `Failed on ${id}`;
+  });
+};
+
+// https://stackoverflow.com/a/46842181/738371
+async function asyncFilter(arr, callback) {
+  const fail = Symbol();
+  return (await Promise.all(arr.map(async item => (await callback(item)) ? item : fail))).filter(i=>i!==fail);
+}
+
+// Returns true where a card is used exactly once, otherwise false.
+async function deletableCard(card, contentfulPreviewConnection) {
+  const cardUsageCount = await contentfulPreviewConnection.getEntries({
+    'links_to_entry': card.sys.id
+  })
+    .then((response) => {
+      return response.items.length;
+    })
+    .catch((e) => {
+      console.log('error retrieving card usage data. \n' + JSON.stringify(e));
+      return 0;
+    });
+  return cardUsageCount === 1;
+}
+
+const getEntryPage = async(contentfulPreviewConnection) => {
+  const entries = await contentfulPreviewConnection.getEntries({
+    'locale': defaultLocale,
+    'content_type': 'imageGallery',
+    'include': 2
+  })
+    .then((response) => {
+      return response.items;
+    })
+    .catch((e) => {
+      console.log('error retrieving gallery data. \n' + JSON.stringify(e));
+    });
+  return entries || [];
+};
+
+const clean = async() => {
+  const contentfulPreviewConnection = await contentfulPreviewClient;
```

Comment:
Move this to a global so that once initialised here it does not need to be passed to the other functions. As we do in other modules for the management connection.